src/main/java/com/rednavis/vaadin/showcase/backend/auth/ConfigureUIServiceInitListener.java

```diff
@@ -0,0 +1,69 @@
+package com.rednavis.vaadin.showcase.backend.auth;
+
+import com.rednavis.vaadin.showcase.backend.auth.annotations.Secured;
+import com.rednavis.vaadin.showcase.backend.auth.jwt.SecurityContext;
+import com.rednavis.vaadin.showcase.exception.ForbiddenError;
+import com.rednavis.vaadin.showcase.views.login.LoginView;
+import com.vaadin.flow.component.UI;
+import com.vaadin.flow.router.BeforeEnterEvent;
+import com.vaadin.flow.server.ServiceInitEvent;
+import com.vaadin.flow.server.VaadinServiceInitListener;
+import java.lang.annotation.Annotation;
+import java.util.stream.Stream;
+import javax.enterprise.context.RequestScoped;
+import javax.inject.Inject;
+import lombok.NoArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * Listener for Vaadin Services initialization events.
+ * It is responsible for checking permissions.
+ */
+@RequestScoped
+@NoArgsConstructor
+@Slf4j
+public class ConfigureUIServiceInitListener implements VaadinServiceInitListener {
+
+  //todo: after checking in debug mode it is unnecessary to use security context with @RequestScoped for storing logged user (Vaadin use several requests for showing page).
+  //todo: So, create hash for storing username, roles and jwt expires date and check user's role in it.
+  @Inject
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use explicit scoping instead of the default package private level](https://app.codacy.com/gh/rednavis/vaadin-showcase/pullRequest?prid=4872368)